core/src/test/java/io/forestframework/core/http/routing/Test405406415.java

```diff
@@ -0,0 +1,297 @@
+        assertThat(getTraces(), equalTo(Collections.singletonList(Message.HANDLER.name())));
+        assertThat(result, containsString(number));
+    }
+
+    @Test
+    void get406WhenProducesAndAcceptsMultiple() throws IOException {
+        // produces = "application/json, text/plain"      accept = "application/xhtml+xml, text/plain; level=2"
+        String result = sendHttpRequest("POST", "/5", "application/xhtml+xml, text/plain;level=2", "text/html").assert406().getBody();
+
+        assertThat(getTraces(), equalTo(Collections.emptyList()));
+        assertThat(result, equalTo(HttpStatusCode.NOT_ACCEPTABLE.name()));
+    }
+
+    @Test
+    void get200WhenProducesAndAcceptsMultiple() throws IOException {
+        // produces = "application/json, text/plain"     accept = "text/html, application/json; q=0.9, text/plain; q=0.8"
+        String result = sendHttpRequest("POST", "/5", "text/html, application/json; q=0.9, text/plain; q=0.8", "text/html").assert200().getBody();
+
+        assertThat(getTraces(), equalTo(Collections.singletonList(Message.HANDLER.name())));
+        assertThat(result, containsString("5"));
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "1, application/*",
+            "1, text/*",
+            "1, text/html",
+            "2, application/*"
+    })
+    void get415ErrorAndHandledByDefaultErrorHandlerWhenConsumesNotMatched(String number, String contentType) throws IOException {
+        // consumes = "application/json"               contentType = "application/*"
+        // consumes = "application/json"               contentType = "text/*"
+        // consumes = "application/json"               contentType = "text/html"
+        // consumes = "application/json, image/jxr"    contentType = "application/*"
+
+        String path = "/" + number;
+        String result = sendHttpRequest("POST", path, "*/*", contentType).assert415().getBody();
+
+        assertThat(getTraces(), equalTo(Collections.emptyList()));
+        assertThat(result, equalTo(HttpStatusCode.UNSUPPORTED_MEDIA_TYPE.name()));
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "3, application/json",
+            "3, text/*",
+    })
+    void get200WhenConsumesMatched(String number, String contentType) throws IOException {
+        // consumes = "*/*"               contentType = "application/json"
+        // consumes = "*/*"               contentType = "text/*"
+        String path = "/" + number;
+        String result = sendHttpRequest("POST", path, "*/*", contentType).assert200().getBody();
+
+        assertThat(getTraces(), equalTo(Collections.singletonList(Message.HANDLER.name())));
+        assertThat(result, containsString(number));
+    }
+
+    @Test
+    void test405ErrorHandledByCustom405ErrorHandler() throws IOException {
+        String path = "/custom/" + "METHOD_NOT_ALLOWED";
+
+        // assertion has to be disabled for now
+        String result = sendHttpRequest("GET", path)
+        //      .assert405()
```

Comment:
Why?

Reply:
This is because `vertx` issue https://github.com/eclipse-vertx/vert.x/issues/3552